pkg/templates/aws/resources/ecr_image.yaml

```diff
@@ -19,19 +32,45 @@ properties:
         unique: true
     description: A reference to the AWS ECR repository resource where the image will
       be stored
+    sanitize: |
+      # https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_CreateRepository.html
+      {{ . 
+        | replace `^[^[:alnum:]]+` "" 
+        | replace `[^[:alnum:]-]+` "-"
+        | replace `-$` ""
+        | length 1 256
+      }}
   Context:
     type: string
     default_value: '.'
     description: The build context path for the Docker build, specifying where to
       find the Dockerfile and other supporting files
+    sanitize: |
+      # No specific AWS documentation, general file path rules applied
+      {{ . 
+        | replace `[^[:alnum:]/.-]+` "" 
```

Comment:
You can definitely have a valid file path that doesn't match these rules, for example the `:` is valid.